modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/QueryEntityValidationSelfTest.java

```diff
@@ -279,4 +290,38 @@ public void testNotUniqueNameInAnnotation() {
             return null;
         }, CacheException.class, "Property with name 'Name3' already exists");
     }
+
+    /**
+     * Test class for sql queryable test key without annotated fields.
+     */
+    private static class TestKeyNoAnnotatedFields {
+        /**
+         * Non-annotated field.
+         */
+        int keyField;
+    }
+
+    /**
+     * Validates a warning is logged when query entity key is potentially misconfigured.
+     */
+    @Test
+    public void testNonAnnotatedField() {
+        CacheConfiguration<TestKeyNoAnnotatedFields, TestValue> ccfg = new CacheConfiguration<>();
+        ccfg.setName(CACHE_NAME);
+        ccfg.setIndexedTypes(TestKeyNoAnnotatedFields.class, TestValue.class);
+
+        LogListener logListener = LogListener.matches("Key of user type has no fields configured for table=TESTVALUE").build();
```

Comment:
I suggest to expand such a message, probably we need to highlight @QuerySqlField  annotation ?

Reply:
Annotation is just one of the ways to configure key fields.